web/ui/src/Services/ServiceDetailsController.js

```diff
@@ -947,8 +955,58 @@
                     return $scope.indent(depth - offset);
                 };
 
-                $scope.setCurrentTreeState = function () {
-                    $scope.serviceTreeState[$scope.currentService.id] = {};
+                $scope.restoreTreeState = function () {
+
+                    if (!$scope.currentService) {
+                        return;
+                    }
+                    // initialize at first call
+                    if (!$scope.serviceTreeState[$scope.currentService.id]) {
+                        $scope.serviceTreeState[$scope.currentService.id] = {};
+                    }
+                    // delete hidden services state upon re-entry
+                    let treeState = $scope.serviceTreeState[$scope.currentService.id];
+                    Object.keys(treeState).forEach(k => {
+                        if (treeState[k].hidden) {
+                            delete treeState[k];
+                        }
+                    });
+                    $scope.serviceTreeState[$scope.currentService.id] = treeState;
+                    
+                    $scope.updateTreeState();
+                    $scope.currentTreeState = treeState;
+
+                };
+
+                $scope.updateTreeState = function () {        
+                    let treeState = $scope.serviceTreeState[$scope.currentService.id];
+                    $scope.currentService.subservices.forEach(topLvlSvc => {
+
+                        (function recurse(service) {
+                            // initialize on page load
+                            if (!treeState[service.id]) {
+                                treeState[service.id] = {};
+                                treeState[service.id].collapsed = true;
+                                treeState[service.id].hidden = false;
```

Comment:
This can more succinctly be written as:

    treeState[service.id] = {
        collapsed: true,
        hidden: false
    };